src/Service.Host/client/src/components/__tests__/BomVerificationHistory.test.js

```diff
@@ -0,0 +1,126 @@
+
+jest.mock('react-redux', () => ({
+    ...jest.requireActual('react-redux'),
+    useSelector: jest.fn()
+}));
+
+jest.mock('react-router', () => ({
+    ...jest.requireActual('react-router'),
+    useLocation: jest.fn()
+}));
+
+const addSpy = jest.spyOn(bomVerificationHistoryActions, 'add');
+const fetchSpy = jest.spyOn(bomVerificationHistoryActions, 'fetch');
+const idParam = {
+    id: 123456
+};
+
+jest.spyOn(routeData, 'useParams').mockReturnValue(idParam);
+
+const initialState = {
+    bomVerificationHistory: {
+        item: null,
+        loading: false
+    }
+};
+
+const bomVerificationHistoryItem = {
+    tRef: 123456,
+    partNumber: 'SK HUB',
+    Remarks: 'B.Slime',
+    verifiedBy: 33084,
+    dateVerified: '2023-01-26T13:55:20.0000000'
+};
+
+const stateWithItem = {
+    bomVerificationHistory: {
+        item: bomVerificationHistoryItem,
+        loading: false
+    }
+};
+
+const stateWithUser = {
+    oidc: { user: { profile: { name: 'User Name', employee: '/employees/33087' } } },
+    parts: {
+        searchItems: [
+            {
+                partNumber: 'TON IC',
+                description: 'A NEW IC',
+                id: 161,
+                bomType: 'C'
+            }
+        ]
+    }
+};
+
+describe('When component mounts...', () => {
+    beforeEach(() => {
+        cleanup();
+        jest.clearAllMocks();
+        useSelector.mockImplementation(callback => callback(initialState));
+        render(<BomVerificationHistory />);
+    });
+
+    test('Should get Bom Ver record', () => {
+        expect(fetchSpy).toBeCalledTimes(1);
+        expect(fetchSpy).toBeCalledWith(123456);
+    });
+});
+
+describe('When item received...', () => {
+    beforeEach(() => {
+        cleanup();
+        jest.clearAllMocks();
+        useSelector.mockImplementation(callback => callback(stateWithItem));
+        render(<BomVerificationHistory />);
+    });
+
+    test('Should render details', () => {
+        expect(screen.getByDisplayValue('SK HUB')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('2023-01-26T13:55:20.0000000')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('123456')).toBeInTheDocument();
+    });
+});
+
+describe('When creating...', () => {
+    beforeEach(() => {
+        cleanup();
+        jest.clearAllMocks();
+        useSelector.mockImplementation(callback => callback(stateWithUser));
+        render(<CreateBomVerificationHistory />);
+    });
+    test('Should add with values on create...', async () => {
+        const searchInput = screen.getByLabelText('Part Number');
+        fireEvent.change(searchInput, { target: { value: 'TON IC' } });
+        fireEvent.keyDown(searchInput, { key: 'Enter', code: 'Enter', keyCode: 13 });
+        const result = screen.getByText('TON IC');
+        fireEvent.click(result);
+
+        // need to wait for modal to close
+        const input = await screen.getByLabelText('Remarks', undefined, { timeout: 5000 });
```

Comment:
you can delete this 'await' keyword and the additional parameters passed to getByLabelText - don't think they're doing anything here

Reply:
Sorted these in a small update during stand up!

Reply:
should delete the comment too